### PR TITLE
Add external_access_integration and secret to python model impl

### PR DESCRIPTION
**Pull request by Lindblomsebastian**
This resolves https://github.com/dbt-labs/dbt-snowflake/issues/925
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/#

### Problem

`EXTERNAL_ACCESS_INTEGRATION` and `SECRETS` are not populated in the generated PROC 

### Solution

Add `EXTERNAL_ACCESS_INTEGRATION` and `SECRETS` to the generated PROC if they are part of the config. 

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by Lindblomsebastian** [comment]
CLA has now been signed.

File: tests/functional/adapter/test_python_model.py [hunk 1]
```diff
@@ -138,3 +138,67 @@ def teardown_method(self, project):
...
+
+
+class TestExternalAccessIntegration:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"external_access_integration_python_model.py": EXTERNAL_ACCESS_INTEGRATION_MODE}
+
+    def test_external_access_integration(self, project):
+        project.run_sql(
+            "create or replace network rule test_network_rule type = host_port mode = egress value_list= ('www.google.com:443');"
```
Comment (colin-rogers-dbt) [inline tests/functional/adapter/test_python_model.py:169]: I don't think we want our integration test account to have these kind of privileges. Let's move these queries to code comments. I'll try to get these set up on our test cluster today

File: tests/functional/adapter/test_python_model.py [hunk 2]
```diff
@@ -138,3 +138,69 @@ def teardown_method(self, project):
...
+
+
+EXTERNAL_ACCESS_INTEGRATION_MODE = """
+import pandas
+import snowflake.snowpark as snowpark
+
+def model(dbt, session: snowpark.Session):
+    dbt.config(
+        materialized="table",
+        external_access_integration=["test_external_access_integration"],
```
Comment (ernestoongaro) [inline tests/functional/adapter/test_python_model.py:150]:
```suggestion
        external_access_integrations=["test_external_access_integration"],
```

**Comment by ernestoongaro** [comment]
@Lindblomsebastian 

I'm getting failures from the secrets test: [dbt.log](https://github.com/dbt-labs/dbt-snowflake/files/14859672/dbt.log)
``` invalid property 'SECRETS' for 'FUNCTION'```

and as well for the external access integrations test: (adding extra quotes) [dbt.log](https://github.com/dbt-labs/dbt-snowflake/files/14859707/dbt.log)
```
  invalid value ''test_external_access_integration'' for property 'EXTERNAL_ACCESS_INTEGRATIONS'
```

**Comment by Lindblomsebastian** [comment]
> @Lindblomsebastian
> 
> I'm getting failures from the secrets test: [dbt.log](https://github.com/dbt-labs/dbt-snowflake/files/14859672/dbt.log) ` invalid property 'SECRETS' for 'FUNCTION'`
> 
> and as well for the external access integrations test: (adding extra quotes) [dbt.log](https://github.com/dbt-labs/dbt-snowflake/files/14859707/dbt.log)
> 
> ```
>   invalid value ''test_external_access_integration'' for property 'EXTERNAL_ACCESS_INTEGRATIONS'
> ```

I missed that the secrets you specify as values must also be specified in the external access integration. (See https://docs.snowflake.com/en/developer-guide/external-network-access/creating-using-external-network-access#using-the-external-access-integration-in-a-function-or-procedure)
@colin-rogers-dbt, would you mind adding a secret named `test_secret` to the `test_external_access_integration`? I've included the SQL commands as comments in the test (See https://github.com/dbt-labs/dbt-snowflake/pull/955/commits/d1517b48e5ddbae411266e0f29e5bd90661b22df#diff-47b9e84b411efd20b1b24f2f1e566455bc7f6dc21021bac6c160efcd76768a60R210-R216).

**Comment by ernestoongaro** [comment]
> @colin-rogers-dbt, would you mind adding a secret named `test_secret` to the `test_external_access_integration`? 

i've done this today but due to the security integration needing to know the schema of the secret I can't quite get it to work so asked ❄️ for help

**Review by ernestoongaro — APPROVED** [review]
Off to @colin-rogers-dbt for review now!

**Review by colin-rogers-dbt — APPROVED** [review]

**Comment by dbeatty10** [comment]
Since this PR introduces a new user interface (model-level config for `external_access_integrations`), I've added the `user docs` label, and I'll ping our documentation team to make sure that they have it tracked.

cc: @dataders

**Comment by colin-rogers-dbt** [comment]
> Since this PR introduces a new user interface (model-level config for `external_access_integrations`), I've added the `user docs` label, and I'll ping our documentation team to make sure that they have it tracked.
> 
> cc: @dataders

Thanks @dbeatty10 !

**Comment by ernestoongaro** [comment]
@dbeatty10 @colin-rogers-dbt - I have a PR for docs: https://github.com/dbt-labs/docs.getdbt.com/pull/5262

**Pull request merged**